std/range/package.d

```diff
@@ -4355,15 +4357,372 @@ if (Ranges.length && allSatisfy!(isInputRange, Ranges))
     }
 }
 
+// Simple formatter defined for code generation in ZipShortest to avoid
+// using std.format in CTFE. Only works for very simple patterns
+// with a single "%s" in `repeatPart`.
+private enum string _zipFormat(string start, string repeatPart, string end, size_t N) =
+    (){
+        import std.string : indexOf;
+        import std.conv : to;
+        immutable subIdx = repeatPart.indexOf("%s");
+        assert(subIdx >= 0);
+        string buf = start;
+        foreach (i; 0 .. N)
+        {
+            if (i != 0) buf ~= ", ";
+            buf ~= repeatPart[0 .. subIdx];
+            buf ~= to!string(i);
+            buf ~= repeatPart[subIdx + 2 .. $];
+        }
+        buf ~= end;
+        return buf;
+    }();
+
+@safe unittest
+{
+    static assert(_zipFormat!("Foo(", "x[%s]", ")", 3) == "Foo(x[0], x[1], x[2])");
+}
+
+/+
+Non-public. Like $(LREF Zip) with `StoppingPolicy.shortest`
+except it properly implements `back` and `popBack` in the
+case of uneven ranges or disables those operations when
+it is not possible to guarantee they are correct.
++/
+package template ZipShortest(Ranges...)
+if (Ranges.length && __traits(compiles, {
+        static assert(allSatisfy!(isInputRange, Ranges));
+    }))
+{
+    alias ZipShortest = .ZipShortest!(
+        Ranges.length == 1 || allSatisfy!(isInfinite, Ranges)
+            ? Yes.allKnownSameLength
+            : No.allKnownSameLength,
+        Ranges);
+}
+/+ non-public, ditto +/
+package struct ZipShortest(Flag!"allKnownSameLength" allKnownSameLength, Ranges...)
+if (Ranges.length && allSatisfy!(isInputRange, Ranges))
+{
+    import std.typecons : Tuple;
+
+    alias R = Ranges; // Unused here but defined in case library users rely on it.
+    private Ranges ranges;
+    alias ElementType = Tuple!(staticMap!(.ElementType, Ranges));
+
+/+
```

Comment:
Please fix the formatting to be more inline with Phobos Function docs should be on the same indentation level as the function.